File: app/src/main/java/com/kaito/quickaction/Google.java
package com.kaito.quickaction;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class Google extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        new Thread(new Runnable() {
            @Override
            public void run() {
                String searchTarget = getIntent().getCharSequenceExtra(Intent.EXTRA_PROCESS_TEXT).toString();
                startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("https://www.google.com/search?q=" + searchTarget)));
            }
        }).start();

        finish();
    }
}

File: app/src/main/java/com/kaito/quickaction/Google.kt
package com.kaito.quickaction

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class Google : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        Thread(Runnable {
            val searchTarget = intent.getCharSequenceExtra(Intent.EXTRA_PROCESS_TEXT).toString()
            startActivity(Intent(Intent.ACTION_VIEW, Uri.parse("https://www.google.com/search?q=$searchTarget")))
        }).start()

        finish()
    }
}
